data/transactions/logic/eval.go

```diff
@@ -629,6 +637,18 @@ func (cx *EvalContext) RunMode() RunMode {
 // PC returns the program counter of the current application being evaluated
 func (cx *EvalContext) PC() int { return cx.pc }
 
+// Stack returns the stack of current cx
+func (cx *EvalContext) Stack() []stackValue { return cx.stack }
```

Comment:
This exposes the underlying array, making it modifiable by callers.  That doesn't bother me all that much, but because it does, I don't think we need to hide it behind an accessor.  Let's just capitalize `cx.Stack`.

Reply:
Well, in this PR we didn't want to modify the stack tho - we just want to read from `stack` from a certain index, to record the before/after effect of an opcode over the stack.

I used `Stack()` to have access to read `cx.stack`, but capitalize it should be fine: see [PR comment from JJ response](https://github.com/algorand/go-algorand/pull/5440/commits/ea9469de34b79818e49e4b41b85d0d53437fbc6d)